openerp/openupgrade/doc/source/development.rst

```diff
@@ -1,7 +1,34 @@
+=========================================================
 Migration script development
-++++++++++++++++++++++++++++
+=========================================================
 
 .. toctree::
    :maxdepth: 2
 
    devfaq
+
+Migration Scrips consist of two (or more) main files in every module's migration directory located in the target source code:
+
+* pre-migration.py (or pre-anythingyouwant.py)
+* post-migration.py (or post-anythingyouwant.py)
+
+What they do is derived from their names, you should check the API Documentation_ of the openupgradelib_ for details of library methods that are readily available for your migration task.
+
+The general purpose of those scripts is to manipulate directly the database entries of the source database in order to change them in a way so that they are compatible with the target database source code. To make that happen smoothly, is your task.
+When you run (TODO DOC HOWTO RUN A CUSTOM MIGRATION) a migration, Odoo's *internal* updating mechanism will be used (OpenUpgrade is providing sugar code for the migration task only). (@Experts: Under what circumstances a failing migration is reverted?)
+
+General recommendations about 'pre-migration.py' and 'post-migration.py'
+------------------------------------------------------------------------
+tbd by an expert
+
+Special Case: Module Renames and Module Merges
+----------------------------------------------
+tbd by an expert
```

Comment:
OK, thanks. Then I encountered other problems, because for us, that did not work. A question for better understanding, so I can try document some internals: At what point Odoo's migration.py is passing on to touch OpenUpgrade code? (Without proper documentation and without clear patch information it's especially hard to figure out what actually happens, even by looking at the code.) I thought for example, that apriori.py is only touched at analysis time.